### PR TITLE
feat: add http

### PR DESCRIPTION
**Pull request by mawkler**
Feel free to change the icon and/or colors. I picked the colors from [this logo on wikipedia](https://en.wikipedia.org/wiki/HTTP?oldformat=true#/media/File:HTTP_logo.svg). I'm not sure how established it is though.

### CODE REVIEW

**Comment by alex-courtis** [comment]
Colours look good dark:


and light:

File: lua/nvim-web-devicons/icons-default.lua [hunk 1]
```diff
@@ -3755,6 +3755,12 @@ local icons_by_operating_system = {
     cterm_color = "250",
     name = "HyperbolaGNULinuxLibre",
   },
+  ["http"] = {
+    icon = "",
+    color = "#008ec7",
+    cterm_color = "38",
+    name = "HTTP",
+  },
```
Comment (alex-courtis) [inline lua/nvim-web-devicons/icons-default.lua:3763]: http is not an operating system.

Please move to `icons_by_file_extension`

**Review by alex-courtis — CHANGES_REQUESTED** [review]
I'm not quite sure of the use case, but this looks good.

Build https://github.com/nvim-tree/nvim-web-devicons/actions/runs/10942703932/job/30409605842?pr=497 is failing due to a light colour mismatch.

The light versions are all generated, please use the tool https://github.com/nvim-tree/nvim-web-devicons/blob/master/CONTRIBUTING.md#generate-colors

**Comment by mawkler** [comment]
@alex-courtis 

> I'm not quite sure of the use case

It is for files containing HTTP requests. They can be used with HTTP clients like [rest.nvim](https://github.com/rest-nvim/rest.nvim) or [kulala.nvim](https://github.com/mistweaverco/kulala.nvim/).

I have now auto generated the light colors 🙂

**Review by alex-courtis — APPROVED** [review]
Many thanks for your contribution!

**Pull request merged**